src/pages/back-to-office/back-to-office.analytics.effects.ts

```diff
@@ -32,4 +38,34 @@ export class BackToOfficeAnalyticsEffects {
       return of();
     }),
   );
+
+  deferWriteUpEffect$ = this.actions$.pipe(
+    ofType(DEFER_WRITE_UP),
+    withLatestFrom(
+      this.store$.pipe(
+        select(getTests),
+        select(getCurrentTest),
+        select(isPassed),
+      ),
+      this.store$.pipe(
+        select(getTests),
+        select(getCurrentTest),
+        select(getJournalData),
+      ),
+    ),
+    switchMap(([action, isPassed, journalDataOfTest]) => {
```

Comment:
Is it worth specifying the type of these arguments? e.g.
`switchMap(([action, isPassed, journalDataOfTest]: [DeferWriteUp, boolean, JournalData]) => {`

Reply:
Can do